base/regex.jl

```diff
@@ -309,6 +309,8 @@ function _replace(io, repl_s::SubstitutionString, str, r, re)
     LBRACKET = '<'
     RBRACKET = '>'
     repl = repl_s.string
+    keep_esc = [SUB_CHAR, GROUP_CHAR, collect('0':'9')...]
```

Comment:
Define this as a literal tuple, or outside of the function with `const`, to avoid allocating on each call.

Reply:
Yes you are right. I fixed that.